qa/src/test/java/io/camunda/migrator/qa/history/entity/HistoryProcessInstanceTest.java

```diff
@@ -0,0 +1,186 @@
+    triggerIncident(c7Process.getId());
+    HistoricProcessInstance historicProcessInstance = historyService.createHistoricProcessInstanceQuery()
+        .processInstanceId(c7Process.getId())
+        .singleResult();
+
+    // when history is migrated
+    historyMigrator.migrate();
+
+    // then expected number of historic process instances
+    List<ProcessInstanceEntity> processInstances = searchHistoricProcessInstances("incidentProcessId");
+    assertThat(processInstances.size()).isEqualTo(1);
+    for (ProcessInstanceEntity processInstance : processInstances) {
+      // and process instance has expected state
+      assertC8ProcessInstanceFields(processInstance, historicProcessInstance, "incidentProcessId",
+          ProcessInstanceEntity.ProcessInstanceState.ACTIVE, null, null, false, true);
+    }
+  }
+
+  @Test
+  @Disabled("https://github.com/camunda/camunda-bpm-platform/issues/5359")
+  public void shouldCheckCalledProcessParentElementKey() {
+
+    // given
+    deployer.deployCamunda7Process("callActivityProcess.bpmn");
+    deployer.deployCamunda7Process("calledActivitySubprocess.bpmn");
+    ProcessInstance parentInstance = runtimeService.startProcessInstanceByKey("callingProcessId");
+    ProcessInstance subInstance = runtimeService.createProcessInstanceQuery()
+        .superProcessInstanceId(parentInstance.getProcessInstanceId())
+        .singleResult();
+    completeAllUserTasksWithDefaultUserTaskId();
+    HistoricActivityInstance callActivity = historyService.createHistoricActivityInstanceQuery()
+        .activityId("callActivityId")
+        .processInstanceId(subInstance.getId())
+        .singleResult();
+    dbClient.insert(callActivity.getId(), null, IdKeyMapper.TYPE.HISTORY_PROCESS_DEFINITION);
+    // when
+    historyMigrator.migrate();
+
+    // then
+    assertThat(searchHistoricProcessInstances("calledProcessInstanceId")).isEmpty();
+  }
+
+  protected static void assertC8ProcessInstanceFields(ProcessInstanceEntity processInstance,
+                                                      HistoricProcessInstance historicProcessInstance,
+                                                      String processDefinitionId,
+                                                      ProcessInstanceEntity.ProcessInstanceState processInstanceState,
+                                                      String versionTag,
+                                                      String tenantId,
+                                                      boolean hasParent,
+                                                      boolean hasIncidents) {
+    assertThat(processInstance.processDefinitionId()).isEqualTo(processDefinitionId);
+    assertThat(processInstance.state()).isEqualTo(processInstanceState);
+    assertThat(processInstance.processInstanceKey()).isNotNull();
+    assertThat(processInstance.processDefinitionKey()).isNotNull();
```

Comment:
Could we also assert that the correct key is referenced? To ensure we test the key lookup/mapping logic involved in the process instance migration

Reply:
I will adjust assert.